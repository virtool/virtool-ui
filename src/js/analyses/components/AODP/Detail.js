import { filter, map, max, mean, min } from "lodash-es";
import React from "react";
import styled from "styled-components";

import { connect } from "react-redux";
import { Badge, BoxGroup, BoxGroupHeader, BoxGroupSection, Icon } from "../../../base";
import { getActiveHit } from "../../selectors";
import { AODPIsolate } from "./Isolate";

const AODPDetailIdentity = styled.div`
    align-items: center;
    display: flex;

    > div:first-child {
        color: ${props => props.theme.color.blue};
        font-size: ${props => props.theme.fontSize.xxl};
    }

    > div:last-child {
        display: flex;
        flex-direction: column;
        font-size: ${props => props.theme.fontSize.sm};
        font-weight: bold;
        justify-content: center;
        margin-left: 5px;

        span:first-child {
            color: ${props => props.theme.color.greenDark};
        }

        span:last-child {
            color: ${props => props.theme.color.redDark};
        }
    }
`;

const AODPDetailOverview = ({ identities }) => {
    return (
        <BoxGroupSection>
            <AODPDetailIdentity>
                <div>{max(identities)}</div>
                <div>
                    <span>~ {mean(identities).toFixed(2)}</span>
                    <span>
                        <Icon name="arrow-down" /> {min(identities)}
                    </span>
                </div>
            </AODPDetailIdentity>
            <div>IDENTITY</div>
        </BoxGroupSection>
    );
};

export const AODPDetail = ({ hit }) => {
    if (hit === null) {
        return <BoxGroup>None found</BoxGroup>;
    }

    const { name, identities, isolates } = hit;

    const filteredIsolates = filter(isolates, isolate => isolate.identities.length);

    const isolateComponents = map(filteredIsolates, isolate => <AODPIsolate key={isolate.id} {...isolate} />);

    return (
        <BoxGroup>
            <BoxGroupHeader>
                <h2>
                    {name} <Badge>{identities.length} hits</Badge>
                </h2>
            </BoxGroupHeader>
            <AODPDetailOverview identities={identities} />
            {isolateComponents}
        </BoxGroup>
    );
};

const mapStateToProps = state => ({
    hit: getActiveHit(state)
});

export default connect(mapStateToProps)(AODPDetail);
