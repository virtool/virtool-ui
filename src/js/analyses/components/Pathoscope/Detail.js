import { filter, map, sum } from "lodash-es";
import React from "react";
import { connect } from "react-redux";
import { ScrollSync } from "react-scroll-sync";
import styled from "styled-components";
import { getActiveHit } from "../../selectors";
import { PathoscopeIsolate } from "./Isolate";

const getContextValue = (isolates, onRendered) => {
    let count = sum(map(isolates, isolate => isolate.sequences.length));

    return {
        count,
        onRendered: () => {
            count--;
            if (count === 0) {
                onRendered();
            }
        }
    };
};

const StyledPathoscopeDetail = styled.div``;

export const PathoscopeDetailContext = React.createContext();

export const PathoscopeDetail = ({ filterIsolates, hit, mappedReads, showPathoscopeReads, onRendered }) => {
    const { isolates, pi } = hit;

    const filtered = filter(isolates, isolate => filterIsolates === false || isolate.pi >= 0.03 * pi);

    const contextValue = getContextValue(filtered, onRendered);

    const isolateComponents = map(filtered, isolate => (
        <PathoscopeIsolate
            key={isolate.id}
            {...isolate}
            reads={Math.round(isolate.pi * mappedReads)}
            showPathoscopeReads={showPathoscopeReads}
        />
    ));

    return (
        <PathoscopeDetailContext.Provider value={contextValue}>
            <StyledPathoscopeDetail>
                <ScrollSync>
                    <div>{isolateComponents}</div>
                </ScrollSync>
            </StyledPathoscopeDetail>
        </PathoscopeDetailContext.Provider>
    );
};

const mapStateToProps = state => ({
    hit: getActiveHit(state),
    filterIsolates: state.analyses.filterIsolates,
    mappedReads: state.analyses.detail.read_count,
    showPathoscopeReads: state.analyses.showPathoscopeReads
});

export default connect(mapStateToProps)(PathoscopeDetail);
