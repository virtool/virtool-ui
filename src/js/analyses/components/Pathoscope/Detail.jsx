import { filter, map, sum } from "lodash-es";
import React from "react";
import { connect } from "react-redux";
import { ScrollSync } from "react-scroll-sync";
import { getActiveHit } from "../../selectors";
import { PathoscopeIsolate } from "./Isolate";

function getContextValue(isolates, onRendered) {
    let count = sum(map(isolates, isolate => isolate.sequences.length));

    return {
        count,
        onRendered: () => {
            count--;
            if (count === 0) {
                onRendered();
            }
        },
    };
}
export const PathoscopeDetailContext = React.createContext();

export const PathoscopeDetail = ({ filterIsolates, hit, mappedReads, showPathoscopeReads, onRendered }) => {
    const { isolates, pi } = hit;

    const filtered = filter(isolates, isolate => filterIsolates === false || isolate.pi >= 0.03 * pi);

    const contextValue = getContextValue(filtered, onRendered);

    const isolateComponents = map(filtered, isolate => (
        <PathoscopeIsolate
            key={isolate.id}
            {...isolate}
            reads={Math.round(isolate.pi * mappedReads)}
            showPathoscopeReads={showPathoscopeReads}
        />
    ));

    return (
        <PathoscopeDetailContext.Provider value={contextValue}>
            <div>
                <ScrollSync>
                    <div>{isolateComponents}</div>
                </ScrollSync>
            </div>
        </PathoscopeDetailContext.Provider>
    );
};

function mapStateToProps(state) {
    return {
        hit: getActiveHit(state),
        filterIsolates: state.analyses.filterIsolates,
        mappedReads: state.analyses.detail.read_count,
        showPathoscopeReads: state.analyses.showPathoscopeReads,
    };
}

export default connect(mapStateToProps)(PathoscopeDetail);
