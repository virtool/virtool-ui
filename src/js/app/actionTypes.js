/**
 * Create a special action type used for requests.
 *
 * The request is an object with three properties assigned with action types used for API requests.
 *
 * - REQUESTED
 * - SUCCEEDED
 * - FAILED
 *
 * @param root {string} the root name of the action type
 * @returns {object} a request-style action type
 */
const createRequestActionType = root => ({
    REQUESTED: `${root}_REQUESTED`,
    SUCCEEDED: `${root}_SUCCEEDED`,
    FAILED: `${root}_FAILED`,
});

// App
export const GET_INITIAL_STATE = createRequestActionType("GET_INITIAL_STATE");

// Account
export const GET_ACCOUNT = createRequestActionType("GET_ACCOUNT");
export const UPDATE_ACCOUNT = createRequestActionType("UPDATE_ACCOUNT");
export const CHANGE_ACCOUNT_PASSWORD = createRequestActionType("CHANGE_ACCOUNT_PASSWORD");
export const LOGIN = createRequestActionType("LOGIN");
export const LOGOUT = createRequestActionType("LOGOUT");
export const RESET_PASSWORD = createRequestActionType("RESET_PASSWORD");

// Administration Settings
export const GET_SETTINGS = createRequestActionType("GET_SETTINGS");
export const UPDATE_SETTINGS = createRequestActionType("UPDATE_SETTINGS");

// Analysis
export const WS_UPDATE_ANALYSIS = "WS_UPDATE_ANALYSIS";
export const SET_ANALYSIS = "SET_ANALYSIS";
export const SET_ANALYSIS_ACTIVE_ID = "SET_ANALYSIS_ACTIVE_ID";
export const SET_SEARCH_IDS = "SET_SEARCH_IDS";
export const SET_AODP_FILTER = "ADD_AODP_FILTER";
export const SET_ANALYSIS_SORT_KEY = "SET_ANALYSIS_SORT_KEY";

// Errors
export const CLEAR_ERROR = "CLEAR_ERROR";

// Files
export const UPLOAD = createRequestActionType("UPLOAD");
export const UPLOAD_SAMPLE_FILE = createRequestActionType("UPLOAD_SAMPLE_FILE");
export const UPLOAD_PROGRESS = "UPLOAD_PROGRESS";
export const UPLOAD_FAILED = "UPLOAD_FAILED";
export const REMOVE_UPLOAD = "REMOVE_UPLOAD";

// Forms
export const SET_PERSISTENT_FORM_STATE = "SET_PERSISTENT_FORM_STATE";
export const DELETE_PERSISTENT_FORM_STATE = "DELETE_PERSISTENT_FORM_STATE";
export const SET_REDUX_FORM_STATE = "SET_REDUX_FORM_STATE";

// Groups
export const CREATE_GROUP = createRequestActionType("CREATE_GROUP");

// HMMs
export const GET_HMM = createRequestActionType("GET_HMM");

// Indexes
export const GET_INDEX = createRequestActionType("GET_INDEX");
export const CREATE_INDEX = createRequestActionType("CREATE_INDEX");

// Jobs
export const GET_JOB = createRequestActionType("GET_JOB");

// OTU
export const GET_OTU = createRequestActionType("GET_OTU");
export const CREATE_OTU = createRequestActionType("CREATE_OTU");
export const EDIT_OTU = createRequestActionType("EDIT_OTU");
export const ADD_ISOLATE = createRequestActionType("ADD_ISOLATE");
export const EDIT_ISOLATE = createRequestActionType("EDIT_ISOLATE");
export const ADD_SEQUENCE = createRequestActionType("ADD_SEQUENCE");
export const EDIT_SEQUENCE = createRequestActionType("EDIT_SEQUENCE");

// Refs
export const GET_REFERENCE = createRequestActionType("GET_REFERENCE");

// Samples
export const GET_SAMPLE = createRequestActionType("GET_SAMPLE");
export const CREATE_SAMPLE = createRequestActionType("CREATE_SAMPLE");
export const UPDATE_SAMPLE = createRequestActionType("UPDATE_SAMPLE");

// Subtraction
export const GET_SUBTRACTION = createRequestActionType("GET_SUBTRACTION");
export const CREATE_SUBTRACTION = createRequestActionType("CREATE_SUBTRACTION");

// Users
export const FIND_USERS = createRequestActionType("FIND_USERS");
export const GET_USER = createRequestActionType("GET_USER");
export const CREATE_USER = createRequestActionType("CREATE_USER");
export const CREATE_FIRST_USER = createRequestActionType("CREATE_FIRST_USER");
export const EDIT_USER = createRequestActionType("EDIT_USER");
