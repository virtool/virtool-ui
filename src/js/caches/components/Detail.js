import React, { useEffect } from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import {
    LoadingPlaceholder,
    RelativeTime,
    SubviewHeader,
    SubviewHeaderAttribution,
    SubviewHeaderTitle
} from "../../base";
import { getCache } from "../actions";
import CacheQuality from "./Quality";
import CacheGeneral from "./General";
import CacheParameters from "./Parameters";

const StyledCacheDetail = styled.div`
    overflow-x: hidden;
`;

export const CacheDetail = ({ detail, match, sampleName, onGet }) => {
    useEffect(() => onGet(match.params.cacheId), [match.params.cacheId]);

    if (detail === null) {
        return <LoadingPlaceholder />;
    }

    return (
        <StyledCacheDetail>
            <SubviewHeader>
                <SubviewHeaderTitle>Cache for {sampleName}</SubviewHeaderTitle>
                <SubviewHeaderAttribution>
                    Created <RelativeTime time={detail.created_at} />
                </SubviewHeaderAttribution>
            </SubviewHeader>
            <CacheGeneral {...detail} />
            <CacheParameters parameters={detail.parameters} />
            <CacheQuality />
        </StyledCacheDetail>
    );
};

export const mapStateToProps = state => {
    return {
        detail: state.caches.detail,
        sampleName: state.samples.detail.name
    };
};

export const mapDispatchToProps = dispatch => {
    return {
        onGet: cacheId => {
            dispatch(getCache(cacheId));
        }
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(CacheDetail);
