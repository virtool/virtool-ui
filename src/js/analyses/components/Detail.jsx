import { get } from "lodash-es";
import React, { useEffect } from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import {
    Box,
    Icon,
    LoadingPlaceholder,
    NotFound,
    RelativeTime,
    SubviewHeader,
    SubviewHeaderAttribution,
    SubviewHeaderTitle,
} from "../../base/index";
import { getWorkflowDisplayName } from "../../utils/utils";
import { getAnalysis } from "../actions";
import AODPViewer from "./AODP/Viewer";
import AnalysisCache from "./CacheLink";
import NuVsViewer from "./NuVs/Viewer";
import { PathoscopeViewer } from "./Pathoscope/Viewer";

const UnsupportedAnalysis = styled(Box)`
    display: flex;
    justify-content: center;
    align-items: center;

    i.fas {
        margin-right: 5px;
    }
`;

export function AnalysisDetail({ detail, error, match, sampleName, onGetAnalysis }) {
    const analysisId = match.params.analysisId;

    useEffect(() => {
        onGetAnalysis(analysisId);
    }, [analysisId]);

    if (error) {
        return <NotFound />;
    }

    if (detail === null || detail.id !== analysisId) {
        return <LoadingPlaceholder />;
    }

    if (!detail.ready) {
        return (
            <Box>
                <LoadingPlaceholder message="Analysis in progress" margin="1.2rem" />
            </Box>
        );
    }

    let content;

    if (detail.workflow === "pathoscope_bowtie") {
        content = <PathoscopeViewer />;
    } else if (detail.workflow === "nuvs") {
        content = <NuVsViewer />;
    } else if (detail.workflow === "aodp") {
        content = <AODPViewer />;
    } else {
        return (
            <>
                <UnsupportedAnalysis>
                    <Icon name={"info-circle"} />
                    Workflow not yet supported.
                </UnsupportedAnalysis>
            </>
        );
    }

    return (
        <div>
            <SubviewHeader>
                <SubviewHeaderTitle>
                    {getWorkflowDisplayName(detail.workflow)} for {sampleName}
                    <AnalysisCache />
                </SubviewHeaderTitle>
                <SubviewHeaderAttribution>
                    {detail.user.handle} started <RelativeTime time={detail.created_at} />
                </SubviewHeaderAttribution>
            </SubviewHeader>

            {content}
        </div>
    );
}

export function mapStateToProps(state) {
    return {
        detail: state.analyses.detail,
        error: get(state, "errors.GET_ANALYSIS_ERROR", null),
        sampleName: state.samples.detail.name,
    };
}

export function mapDispatchToProps(dispatch) {
    return {
        onGetAnalysis: analysisId => {
            dispatch(getAnalysis(analysisId));
        },
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(AnalysisDetail);
