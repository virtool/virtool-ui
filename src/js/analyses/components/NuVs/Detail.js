import { filter, map, sortBy } from "lodash-es";
import React from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { getBorder } from "../../../app/theme";
import { Badge, Box } from "../../../base";
import { useElementSize } from "../../../utils/hooks";
import { getActiveHit, getMaxSequenceLength } from "../../selectors";
import NuVsBLAST from "./BLAST";
import { NuVsORF } from "./ORF";
import { NuVsSequence } from "./Sequence";
import { NuVsValues } from "./Values";

const StyledNuVsFamilies = styled.div`
    border: ${getBorder};
    border-radius: ${props => props.theme.borderRadius.sm};
    display: flex;
    margin: 10px 0 5px;
    overflow: hidden;

    div {
        padding: 5px 10px;
    }

    div:first-child {
        background-color: ${props => props.theme.color.greyLightest};
        border-right: ${getBorder};
    }
`;

const NuVsFamilies = ({ families }) => (
    <StyledNuVsFamilies>
        <div>Families</div>
        <div>{families.length ? families.join(", ") : "None"}</div>
    </StyledNuVsFamilies>
);

const NuVsLayout = styled.div`
    border: ${getBorder};
    margin-bottom: 15px;

    & > div:nth-child(even) {
        background-color: ${props => props.theme.color.greyLightest};
    }
`;

const NuVsDetailTitle = styled.div`
    margin-bottom: 10px;

    h3 {
        align-items: center;
        display: flex;
        font-size: ${props => props.theme.fontSize.lg};
        font-weight: bold;
        justify-content: space-between;
        margin: 0;
    }

    span {
        font-size: ${props => props.theme.fontSize.md};
        font-weight: bold;
    }

    ${Badge} {
        font-size: ${props => props.theme.fontSize.md};
        padding: 5px 10px;
    }
`;

const StyledNuVsDetail = styled(Box)`
    align-items: stretch;
    display: flex;
    flex-direction: column;
    min-height: 500px;
    min-width: 0;
    margin-left: 10px;
`;

export const NuVsDetail = ({ filterORFs, hit, maxSequenceLength }) => {
    const [ref, { width }] = useElementSize();

    if (!hit) {
        return <StyledNuVsDetail>No Hits</StyledNuVsDetail>;
    }

    const { e, families, index, orfs, sequence } = hit;

    let filtered;

    if (filterORFs) {
        filtered = filter(orfs, orf => orf.hits.length);
    }

    filtered = sortBy(filtered || orfs, orf => orf.hits.length).reverse();

    const orfComponents = map(filtered, (orf, index) => (
        <NuVsORF key={index} index={index} {...orf} maxSequenceLength={maxSequenceLength} width={width} />
    ));

    return (
        <StyledNuVsDetail ref={ref}>
            <NuVsDetailTitle>
                <h3>
                    Sequence {index}
                    <Badge>{sequence.length} bp</Badge>
                </h3>
                <NuVsValues e={e} orfCount={orfs.length} />
                <NuVsFamilies families={families} />
            </NuVsDetailTitle>
            <NuVsLayout>
                <NuVsSequence key="sequence" maxSequenceLength={maxSequenceLength} sequence={sequence} width={width} />
                {orfComponents}
            </NuVsLayout>
            <NuVsBLAST />
        </StyledNuVsDetail>
    );
};

const mapStateToProps = state => ({
    filterORFs: state.analyses.filterORFs,
    maxSequenceLength: getMaxSequenceLength(state),
    hit: getActiveHit(state)
});

export default connect(mapStateToProps)(NuVsDetail);
