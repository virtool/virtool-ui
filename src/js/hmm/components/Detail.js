import { get, map } from "lodash-es";
import React from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { getBorder } from "../../app/theme";

import {
    device,
    Badge,
    BoxGroup,
    BoxGroupHeader,
    Label,
    LoadingPlaceholder,
    NotFound,
    Table,
    ViewHeader,
    ViewHeaderTitle
} from "../../base";
import { getHmm } from "../actions";
import { ClusterMember } from "./ClusterMember";
import { HMMTaxonomy } from "./Taxonomy";

const ClusterTable = styled(Table)`
    border: none;
    display: flex;
    flex-flow: column;
    height: 330px !important;
    margin: 0;
    width: 100%;

    thead {
        flex: 0 0 auto;
        width: calc(100% - 0.9em);

        th {
            border-bottom: none;
        }
    }

    tbody {
        flex: 1 1 auto;
        display: block;
        overflow-y: auto;
        border-top: ${getBorder};

        tr {
            width: 100%;
        }

        tr:first-child td {
            border-top: none;
        }
    }

    thead,
    tbody tr {
        display: table;
        table-layout: fixed;
    }
`;

const TaxonomyGrid = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: ${props => props.theme.gap.column};

    @media (max-width: ${device.tablet}) {
        grid-template-columns: 1fr;
    }
`;

export class HMMDetail extends React.Component {
    componentDidMount() {
        this.props.onGet(this.props.match.params.hmmId);
    }

    render() {
        if (this.props.error) {
            return <NotFound />;
        }

        if (this.props.detail === null) {
            return <LoadingPlaceholder margin="130px" />;
        }
        const clusterMembers = map(this.props.detail.entries, ({ name, accession, organism }, index) => (
            <ClusterMember name={name} accession={accession} organism={organism} key={index} />
        ));

        const names = map(this.props.detail.names, (name, index) => (
            <Label key={index} spaced>
                {name}
            </Label>
        ));

        const title = this.props.detail.names[0];

        return (
            <div>
                <ViewHeader title={title}>
                    <ViewHeaderTitle>{title}</ViewHeaderTitle>
                </ViewHeader>

                <BoxGroup>
                    <BoxGroupHeader>
                        <h2>General</h2>
                    </BoxGroupHeader>

                    <Table>
                        <tbody>
                            <tr>
                                <th>Cluster</th>
                                <td>{this.props.detail.cluster}</td>
                            </tr>

                            <tr>
                                <th>Best Definitions</th>
                                <td>{names}</td>
                            </tr>

                            <tr>
                                <th>Length</th>
                                <td>{this.props.detail.length}</td>
                            </tr>

                            <tr>
                                <th>Mean Entropy</th>
                                <td>{this.props.detail.mean_entropy}</td>
                            </tr>

                            <tr>
                                <th>Total Entropy</th>
                                <td>{this.props.detail.total_entropy}</td>
                            </tr>
                        </tbody>
                    </Table>
                </BoxGroup>

                <BoxGroup>
                    <BoxGroupHeader>
                        <h2>
                            Cluster Members <Badge>{this.props.detail.entries.length}</Badge>
                        </h2>
                    </BoxGroupHeader>
                    <ClusterTable>
                        <thead>
                            <tr>
                                <th>Accession</th>
                                <th>Name</th>
                                <th>Organism</th>
                            </tr>
                        </thead>
                        <tbody>{clusterMembers}</tbody>
                    </ClusterTable>
                </BoxGroup>

                <TaxonomyGrid>
                    <HMMTaxonomy title="Families" counts={this.props.detail.families} />
                    <HMMTaxonomy title="Genera" counts={this.props.detail.genera} />
                </TaxonomyGrid>
            </div>
        );
    }
}

export const mapStateToProps = state => ({
    error: get(state, "errors.GET_HMM_ERROR", null),
    detail: state.hmms.detail
});

export const mapDispatchToProps = dispatch => ({
    onGet: hmmId => {
        dispatch(getHmm(hmmId));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(HMMDetail);
