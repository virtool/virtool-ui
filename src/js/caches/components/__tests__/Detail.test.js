import { CacheDetail, mapStateToProps, mapDispatchToProps } from "../Detail";
import { GET_CACHE } from "../../../app/actionTypes";

describe("<CacheDetail />", () => {
    let props;

    beforeEach(() => {
        props = {
            detail: {
                hash: "123abc",
                program: "foo",
                created_at: "2018-01-01T00:00:00.000000Z",
                parameters: {
                    foo: "bar"
                }
            },
            match: {
                params: {
                    cacheId: "baz"
                }
            },
            sampleName: "Foo",
            onGet: jest.fn()
        };
    });

    it("should render", () => {
        const wrapper = shallow(<CacheDetail {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render when [detail=null]", () => {
        props.detail = null;
        const wrapper = shallow(<CacheDetail {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("mapStateToProps()", () => {
    it("should return props", () => {
        const state = {
            caches: {
                detail: "foo"
            },

            samples: {
                detail: {
                    name: "bar"
                }
            }
        };

        const props = mapStateToProps(state);
        expect(props).toEqual({
            detail: "foo",
            sampleName: "bar"
        });
    });
});

describe("mapDispatchToProps", () => {
    it("should return onGet() in props", () => {
        const dispatch = jest.fn();
        const props = mapDispatchToProps(dispatch);

        props.onGet("foo");
        expect(dispatch).toHaveBeenCalledWith({
            type: GET_CACHE.REQUESTED,
            cacheId: "foo"
        });
    });
});
