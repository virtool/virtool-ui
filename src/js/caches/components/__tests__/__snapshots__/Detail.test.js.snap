// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CacheDetail /> should render 1`] = `
<Detail__StyledCacheDetail>
  <ViewHeader__SubviewHeader>
    <ViewHeader__SubviewHeaderTitle>
      Cache for 
      Foo
    </ViewHeader__SubviewHeaderTitle>
    <ViewHeader__SubviewHeaderAttribution>
      Created 
      <RelativeTime
        time="2018-01-01T00:00:00.000000Z"
      />
    </ViewHeader__SubviewHeaderAttribution>
  </ViewHeader__SubviewHeader>
  <CacheGeneral
    created_at="2018-01-01T00:00:00.000000Z"
    hash="123abc"
    parameters={
      Object {
        "foo": "bar",
      }
    }
    program="foo"
  />
  <CacheParameters
    parameters={
      Object {
        "foo": "bar",
      }
    }
  />
  <Connect(CacheQuality) />
</Detail__StyledCacheDetail>
`;

exports[`<CacheDetail /> should render when [detail=null] 1`] = `<LoadingPlaceholder />`;
